gpMgmt/bin/gppylib/operations/package.py

```diff
@@ -1637,3 +1617,29 @@ def execute(self):
         ParallelOperation(operations).run()
         for operation in operations:
             operation.get_ret()
+
+
+# As of py3.8, platform library doesn't have linux_distribution.
+# But all the production systems we support use systemd and hence have this info in /etc/os-release.
+# https://www.freedesktop.org/software/systemd/man/os-release.html
+def _platform_linux_distribution(search_key, osid_filename):
+    try:
+        with open(osid_filename) as f:
+            full_key = '%s=' % search_key
```

Comment:
super minor nit: I find it easier to keep track of less things mentally in a function, especially for small stuff. Thus, simply using `search_key` instead of `full_key`. 
```suggestion
            search_key = '%s=' % search_key
```

Reply:
This then modifies the formal parameter.  Leaving it as is.